### PR TITLE
DisjointUnionSet | Trie | Java

### PR DESCRIPTION
**Pull request by sonu7089**
Created DisjointSet Implementation in Java
and Create Simple and Clean Trie Implementation for Trie in Java
For issue #317 
and For issue #314 
Pr will be closed and marked as spam. If the issue number not found or the issue was assigned to someone else. 
Marking as spam can block your account from HacktoberFest.
### Self Check(Tick After Making pull Request)

- [x] This issue was assigned to me if not it will be marked as spam.
- [x] My file is in the proper folder.
- [x] I am following clean code and Documentation.
- [x] I have added the title and what program will do first in the file.

### CODE REVIEW

File: Java/Data-Structures/DSU/DisjointUnionSet.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    }
+    public boolean union(int x, int y) {
+        int a = find(x);
+        int b = find(y);
+        if(a == b)
+            return false;
+        if(size[a] < size[b]){
+            parent[a] = parent[b];
+            size[b] += size[a];
+        }else{
```
Comment (ahmeddrawy) [inline Java/Data-Structures/DSU/DisjointUnionSet.java:31]:
```suggestion
        }else{
        /// if the size of B subtree smaller than the b one we should merge the B to A
        /// so we make the parent of B is the A parent and increase the size of A subtree by the size of B subtree
```

**Review by ahmeddrawy — CHANGES_REQUESTED** [review]
Overall I don't understand the usage of the `count`  in  `DSU` class and added some comments to make the code easily understandable
The way to use these classes still not obvious and need to be documented and have more comments and define the inputs and outputs constraints and structure

File: Java/Data-Structures/DSU/DisjointUnionSet.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            parent[b] = parent[a];
+            size[a] += size[b];
+        }
+        count--;
```
[the remark this answers is not in the document]
Comment (sonu7089) [reply Java/Data-Structures/DSU/DisjointUnionSet.java:35]: The Idea was whenever we are connecting two components that were not previously connected we are reducing the total number of connected components in the disjoint set.
Will add comments for a better understanding

[the remark this answers is not in the document]
Comment (ahmeddrawy) [reply Java/Data-Structures/DSU/DisjointUnionSet.java:40]: thank you, I see the `count` here has no use, you can add utility functions using DSU to count number of connected components for example where the count may make sense but here I found it has no actual use

[the remark this answers is not in the document]
Comment (sonu7089) [reply Java/Data-Structures/DSU/DisjointUnionSet.java:40]: Yeah but when we are making thing more in general we are actually making sense although using utility function will be slow for big size of disjoint set, it is better and faster.
Moreover you can add the utility function here so we can see which helps us solve the problem to count components faster.

**Pull request merged**